perun-core/src/main/java/cz/metacentrum/perun/core/impl/VosManagerImpl.java

```diff
@@ -158,20 +158,14 @@ public Vo createVo(PerunSession sess, Vo vo) throws VoExistsException {
 		if (this.voExists(sess, vo)) throw new VoExistsException(vo.toString());
 		if (this.shortNameForVoExists(sess, vo)) throw new VoExistsException(vo.toString());
 
-		// Get VO ID
-		int voId;
 		try {
-			voId = Utils.getNewId(jdbc, "vos_id_seq");
-			jdbc.update("insert into vos(id, name, short_name, created_by,modified_by, created_by_uid, modified_by_uid) values (?,?,?,?,?,?,?)",
-					voId, vo.getName(), vo.getShortName(), sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getUserId(), sess.getPerunPrincipal().getUserId());
+			// Get VO ID
+			int voId = Utils.getNewId(jdbc, "vos_id_seq");
+			return jdbc.queryForObject("insert into vos(id, name, short_name, created_by,modified_by, created_by_uid, modified_by_uid) values (?,?,?,?,?,?,?) returning " + voMappingSelectQuery,
```

Comment:
We should replace usage of `Utils.getNewId(jdbc,"seq_name")` with direct SQL call within insert statement, like: `...values (nextval('vos_id_seq'),?,?,.....`

Reply:
Done